### PR TITLE
Add integration tests for actions

### PR DESCRIPTION
**Pull request by Aflynn50**
Add integration test for actions. The test deploys a very basic charm and runs some actions against it. It checks:
- A successful action 
- An action marked as failed by the charm
- An action with incorrect parameters, rejected by juju cli

These tests are motivated by https://github.com/juju/charm/pull/424 which disallows extra parameters for actions.

I have also updated the comments in some of the test scripts because they were not very useful.
## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [x] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing
- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

`./main.sh -v actions`

## Links

**Jira card:** JUJU-5501

### CODE REVIEW

File: testcharms/charm-hub/charms/actions/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Actions charm
+
+A charm used in the actions integration tests. I has an action called `fortune`, see charmcraft.yaml for more.
```
Comment (tonyandrewmeyer) [inline testcharms/charm-hub/charms/actions/README.md:3]:
```suggestion
A charm used in the actions integration tests. I have an action called `fortune`, see charmcraft.yaml for more.
```

(Unless this was lol speak on purpose).

File: testcharms/charm-hub/charms/actions/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+ops ~= 2.5
```
Comment (tonyandrewmeyer) [inline testcharms/charm-hub/charms/actions/requirements.txt:1]: Any reason not to make this 2.10?

**Review by tonyandrewmeyer — COMMENTED** [review]
It seems like it would be worth also having an action where `additionalProperties` is true to verify that behaviour still works.

File: testcharms/charm-hub/charms/actions/README.md (hunk 1 above)
Comment (Aflynn50) [reply testcharms/charm-hub/charms/actions/README.md]: The new official Canonical writing guidelines state British English is to be replaced by lol speak where possible.

I've changed it to "It has" :)

File: testcharms/charm-hub/charms/actions/requirements.txt (hunk 2 above)
Comment (Aflynn50) [reply testcharms/charm-hub/charms/actions/requirements.txt]: Good point, changed.

**Comment by Aflynn50** [comment]
> It seems like it would be worth also having an action where `additionalProperties` is true to verify that behaviour still works.

Good point, I've added an extra with `additionalProperties: true` action that returns all the parameters it gets.

Thanks for the review!

**Review by tonyandrewmeyer — APPROVED** [review]
Looks great, thanks!

(After I suggested using ops 2.10 I annoyingly released 2.11, but I think 2.10 is fine).

File: tests/suites/actions/run_actions.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	juju run juju-qa-action/0 fortune length="long"
+	# Check the the task succeeded.
+	juju show-operation 1 --format=json | jq '.status' | check 'completed'
+	# Check the action succeeded.
+	juju show-task 2 --format=json | jq '.status' | check 'completed'
+
+	# Run an action that will return failed status.
+	juju run juju-qa-action/0 fortune length="long" fail="fail with this string"
+	# Check the the task succeeded.
+	juju show-operation 3 --format=json | jq '.status' | check 'failed'
```
Comment (SimonRichardson) [inline tests/suites/actions/run_actions.sh:25]: The comment is wrong, as the check failed (task failed successfully?)

Comment (Aflynn50) [reply tests/suites/actions/run_actions.sh:25]: Changed

**Review by jack-w-shaw — APPROVED** [review]

**Pull request merged**